raiden/transfer/token_network.py

```diff
@@ -30,50 +30,78 @@
 
+
+def subdispatch_to_channels(
+    block_hash: BlockHash,
+    block_number: BlockNumber,
+    channel_states: List[NettingChannelState],
+    state_change: StateChangeWithChannelID,
+    token_network_state: TokenNetworkState
+) -> TransitionResult:
+    events = []
     ids_to_channels = token_network_state.channelidentifiers_to_channels
+    for channel_state in channel_states:
+        result = channel.state_transition(
+            channel_state=channel_state,
+            state_change=state_change,
+            block_number=block_number,
+            block_hash=block_hash,
+        )
 
-    channel_state = None
-    if node_address is not None:
-        if node_address in ids_to_channels:
-            channel_state = ids_to_channels[node_address].get(state_change.channel_identifier)
+        partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
+            channel_state.partner_state.address
+        ]
+
+        channel_identifier = state_change.channel_identifier
+        if result.new_state is None:
+            del ids_to_channels[channel_state.our_state.address][channel_identifier]
+            partner_to_channelids.remove(channel_identifier)
         else:
-            lc_address = views.get_lc_address_by_channel_id_and_partner(token_network_state, node_address,
-                                                                    state_change.canonical_identifier)
-            node_address = lc_address
-            if lc_address in token_network_state.channelidentifiers_to_channels:
-                channel_state = token_network_state.channelidentifiers_to_channels[lc_address].get(
-                    state_change.canonical_identifier.channel_identifier)
-        if channel_state:
-            result = channel.state_transition(
-                channel_state=channel_state,
-                state_change=state_change,
-                block_number=block_number,
-                block_hash=block_hash,
-            )
-
-            partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
-                channel_state.partner_state.address
-            ]
+            ids_to_channels[channel_state.our_state.address][channel_identifier] = result.new_state
 
-            channel_identifier = state_change.channel_identifier
-            if result.new_state is None:
-                del ids_to_channels[node_address][channel_identifier]
-                partner_to_channelids.remove(channel_identifier)
-            else:
-                ids_to_channels[node_address][channel_identifier] = result.new_state
+        events.extend(result.events)
+    return TransitionResult(token_network_state, events)
 
-            events.extend(result.events)
 
-    return TransitionResult(token_network_state, events)
+#  TODO: We should find another approach for ths function. It's "a bit" confusing
```

Comment:
either we deal with this or we remove it.

if necessary, please create a ticket. 

there is not much added value for this comment, i think.